### PR TITLE
Support specifying trimMantissa with the old string format

### PR DESCRIPTION
**Pull request by lexanth**
Currently there is no way to convey `trimMantissa` when using the old string format.

Numeral uses square brackets around decimals to convey that these are optional, so use that.

A more complete change would be extending numbro to support a min and max range of mantissa values, but that seems like a much more involved change and this would unblock our migration from numeral to numbro.

### CODE REVIEW

File: src/parsing.js [hunk 1]
```diff
@@ -172,6 +172,7 @@ function parseMantissa(string, result) {
         let match = mantissa.match(/0+/);
         if (match) {
             result.mantissa = match[0].length;
+            result.trimMantissa = mantissa.indexOf("[") !== -1;
```
Comment (BenjaminVanRyseghem) [inline src/parsing.js:175]: could you please extract this in a separate method please?

I would like to keep a one-to-one option/parsing function to keep things more separated :smile:

**Review by BenjaminVanRyseghem — CHANGES_REQUESTED** [review]
Thanks a lot :smile:

Could you please add yourself in the AUTHORS file, as well as add a line in CHANGELOG.md?

**Comment by lexanth** [comment]
> Thanks a lot 😄
> 
> Could you please add yourself in the AUTHORS file, as well as add a line in CHANGELOG.md?

@BenjaminVanRyseghem all done!

**Review by BenjaminVanRyseghem — APPROVED** [review]

**Comment by BenjaminVanRyseghem** [comment]
thanks a lot!

@all-contributors please add @lexanth  for code and bug

**Comment by BenjaminVanRyseghem** [comment]
Adds @lexanth as a contributor for code, bug.

**Comment by lexanth** [comment]
@BenjaminVanRyseghem any idea when you're going to do another release with this included?

**Pull request merged**